File: FruitWholesale.Client/src/app/features/supplier-payment/supplier-payment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { PaginatedList, PaginationRequest } from '../../core/models/common.model';
import { SaveSupplierPayment, SupplierPayment } from '../../core/models/transactions.model';
import { toHttpParams } from '../../core/utils/http-params.util';

@Injectable({ providedIn: 'root' })
export class SupplierPaymentService {
  private readonly baseUrl = `${environment.apiUrl}/supplierpayment`;

  constructor(private readonly http: HttpClient) {}

  getPaged(request: PaginationRequest, supplierId?: number | null, fromDate?: string | null, toDate?: string | null) {
    return this.http.get<PaginatedList<SupplierPayment>>(this.baseUrl, {
      params: toHttpParams({ ...request, supplierId, fromDate, toDate })
    });
  }

  create(dto: SaveSupplierPayment): Observable<SupplierPayment> {
    return this.http.post<SupplierPayment>(this.baseUrl, dto);
  }

  update(id: number, dto: SaveSupplierPayment): Observable<SupplierPayment> {
    return this.http.put<SupplierPayment>(`${this.baseUrl}/${id}`, dto);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}

File: FruitWholesale.Mobile/lib/features/supplier_payment/supplier_payment_service.dart
import '../../core/api/api_client.dart';
import '../../core/models/paginated_list.dart';
import 'supplier_payment_models.dart';

class SupplierPaymentService {
  final ApiClient _api;

  SupplierPaymentService(this._api);

  Future<PaginatedList<SupplierPayment>> getPaged({
    int pageNumber = 1,
    int pageSize = 20,
    int? supplierId,
    String? fromDate,
    String? toDate,
  }) async {
    final json = await _api.get('/supplierpayment', query: {
      'pageNumber': pageNumber,
      'pageSize': pageSize,
      if (supplierId != null) 'supplierId': supplierId,
      if (fromDate != null) 'fromDate': fromDate,
      if (toDate != null) 'toDate': toDate,
    });
    return PaginatedList.fromJson(json as Map<String, dynamic>, SupplierPayment.fromJson);
  }

  Future<SupplierPayment> getById(int id) async {
    final json = await _api.get('/supplierpayment/$id');
    return SupplierPayment.fromJson(json as Map<String, dynamic>);
  }

  Future<SupplierPayment> create(SupplierPayment payment) async {
    final json = await _api.post('/supplierpayment', body: payment.toSaveJson());
    return SupplierPayment.fromJson(json as Map<String, dynamic>);
  }

  Future<SupplierPayment> update(int id, SupplierPayment payment) async {
    final json = await _api.put('/supplierpayment/$id', body: payment.toSaveJson());
    return SupplierPayment.fromJson(json as Map<String, dynamic>);
  }
}
